docs/clients/service-providers.md

```diff
@@ -0,0 +1,103 @@
+
+### Command-Line interface
+
+Next you will find a few useful CLI commands to interact with the Full-Node.
+
+#### Creating a key-pair
+
+To generate a new key (default ed25519 elliptic curve):
+
+```bash
+gaiacli keys add <your_key_name>
+```
+
+You will be asked to create a passwords (at least 8 characters) for this key-pair. The command returns 4 informations:
+
+- `NAME`: Name of your key
+- `ADDRESS`: Your address. Used to receive funds.
+- `PUBKEY`: Your public key. Useful for validators.
+- `Seed phrase`: 12-words phrase. **Save this seed phrase somewhere safe**. It is used to recover your private key in case you forget the password.
+
+You can see all your available keys by typing:
+
+```bash
+gaiacli keys list
+```
+
+#### Checking your balance
+
+After receiving tokens to your address, you can view your account's balance by typing:
+
+```bash
+gaiacli account <YOUR_ADDRESS>
+```
+
+*Note: When you query an account balance with zero tokens, you will get this error: No account with address <YOUR_ADDRESS> was found in the state. This is expected! We're working on improving our error messages.*
+
+#### Sending coins via the CLI
+
+Here is the command to send coins via the CLI:
+
+```bash
+gaiacli send --amount=10faucetToken --chain-id=<name_of_testnet_chain> --name=<key_name> --to=<destination_address>
+```
+
+Flags:
+- `--amount`: This flag accepts the format `<value|coinName>`.
+- `--chain-id`: This flag allows you to specify the id of the chain. There will be different ids for different testnet chains and main chain.
+- `--name`: Name of the key of the sending account.
+- `--to`: Address of the recipient.
+
+
+
+## Setting up the Rest Server
+
+The Rest Server acts as an intermediary between the front-end and the full-node. You don't need to run the Rest Server on the same machine as the full-node. If you intend to run the Rest Server on another machine, you need to go through the [Installation and configuration](#installation-and-configuration) again on this machine.
+
+To start the Rest server: 
+
+```bash
+gaiacli advanced rest-server --trust-node=false --node=<full_node_address:full_node_port>
+```
+
+Flags:
+- `--trust-node`: A boolean. If `true`, light-client verification is enabled. If `false`, it is disabled. For service providers, this should be set to `false`.
+- `--node`: This is where you indicate the address and the port of your full-node. The format is <full_node_address:full_node_port>. If the full-node is on the same machine, the address should be "tcp://localhost".
+- `--laddr`: This flag allows you to specify the address and port for the Rest Server. You will mostly use this flag only to specify the port, in which case just input "localhost" for the address. The format is <rest_server_address:port>.
+
+### Listening for incoming transaction
+
+The current recommended way to listen for incoming transaction is to periodically query the blockchain through the following endpoint of the LCD:
```

Comment:
The ~current~ recommended way to listen for incoming transaction~s~ is to periodically query the blockchain through the following endpoint of the LCD:

Reply:
is this true with the new implementation of queriers? @faboweb ?